src/mappings/bondingManager.ts

```diff
@@ -211,6 +211,9 @@ export function transferBond(event: TransferBond): void {
   transferBondEvent.newUnbondingLockId = event.params.newUnbondingLockId.toI32();
   transferBondEvent.oldUnbondingLockId = event.params.oldUnbondingLockId.toI32();
   transferBondEvent.save();
+
+  // Remove UnbondEvent as the unbondingLock is immediately processed and event is no longer valid
+  store.remove("UnbondEvent", makeEventId(event.transaction.hash, event.logIndex.minus(BigInt.fromI32(1))));
```

Comment:
since TransferBond event is emitted immediately after Unbond event, the log index for the Unbond event could be derived by subtracting 1 from the TransferBond event's index.